src/main/scala/org/apache/spark/deploy/armada/submit/ArmadaClientApplication.scala

```diff
@@ -211,42 +187,311 @@ private[spark] class ArmadaClientApplication extends SparkApplication {
+    * @param podLabels
+    *   Kubernetes labels applied to all pods (driver and executors)
+    * @param driverLabels
+    *   Additional Kubernetes labels applied only to driver pod
+    * @param executorLabels
+    *   Additional Kubernetes labels applied only to executor pods
+    * @param armadaClusterUrl
+    *   URL for connecting to Armada cluster
+    * @param nodeSelectors
+    *   Kubernetes node selectors for pod placement
+    * @param nodeUniformityLabel
+    *   Label key for gang scheduling node uniformity constraints
+    * @param executorConnectionTimeout
+    *   Maximum time to wait for executor connection to driver
+    * @param driverLimitCores
+    *   Hard CPU limit for the driver pod
+    * @param driverRequestCores
+    *   CPU request for the driver pod
+    * @param driverLimitMemory
+    *   Hard memory limit for the driver pod
+    * @param driverRequestMemory
+    *   Memory request for the driver pod
+    * @param executorLimitCores
+    *   Hard CPU limit for each executor pod
+    * @param executorRequestCores
+    *   CPU request for each executor pod
+    * @param executorLimitMemory
+    *   Hard memory limit for each executor pod
+    * @param executorRequestMemory
+    *   Memory request for each executor pod
+    */
+  private[spark] case class CLIConfig(
+      queue: Option[String],
+      jobSetId: Option[String],
+      namespace: Option[String],
+      priority: Option[Double],
+      containerImage: Option[String],
       podLabels: Map[String, String],
       driverLabels: Map[String, String],
       executorLabels: Map[String, String],
-      armadaClusterUrl: String,
+      armadaClusterUrl: Option[String],
       nodeSelectors: Map[String, String],
       nodeUniformityLabel: Option[String],
-      executorConnectionTimeout: Duration
+      executorConnectionTimeout: Option[Duration],
+      driverLimitCores: Option[String],
+      driverRequestCores: Option[String],
+      driverLimitMemory: Option[String],
+      driverRequestMemory: Option[String],
+      executorLimitCores: Option[String],
+      executorRequestCores: Option[String],
+      executorLimitMemory: Option[String],
+      executorRequestMemory: Option[String]
+  )
+
+  /** Configuration object for Armada job submission parameters.
+    *
+    * @param jobTemplate
+    *   Optional loaded job template for advanced job customization
+    * @param driverJobItemTemplate
+    *   Optional loaded driver job item template
+    * @param executorJobItemTemplate
+    *   Optional loaded executor job item template
+    * @param cliConfig
+    *   CLI configuration parameters parsed from --conf options
+    */
+  private[spark] case class ArmadaJobConfig(
+      jobTemplate: Option[api.submit.JobSubmitRequest],
+      driverJobItemTemplate: Option[api.submit.JobSubmitRequestItem],
+      executorJobItemTemplate: Option[api.submit.JobSubmitRequestItem],
+      cliConfig: CLIConfig
   )
 
   private def submitArmadaJob(
```

Comment:
This is now a fairly large method and kind of hard to follow.   What do you think of splitting it into 3 or 4 methods?

Reply:
refactored